backend/src/main/java/se/chalmers/dat265/group1/api/physical/PhysicalsController.java

```diff
@@ -0,0 +1,91 @@
+package se.chalmers.dat265.group1.api.physical;
+
+import se.chalmers.dat265.group1.api.ApiController;
+import se.chalmers.dat265.group1.model.PhysicalPart;
+import se.chalmers.dat265.group1.model.PhysicalPrint;
+import se.chalmers.dat265.group1.model.SlmData;
+import se.chalmers.dat265.group1.model.dto.PhysicalPrintSlm;
+import se.chalmers.dat265.group1.storage.FileUtil;
+
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class PhysicalsController extends ApiController implements PhysicalAPI {
+    public PhysicalsController(boolean debug) {
+        super(debug);
+    }
+
+    @Override
+    public List<PhysicalPart> getAllPhysicalParts() {
+        return physicalPartRepository.getObjects();
+    }
+
+    @Override
+    public PhysicalPart getPhysicalPart(String physicalPartID) {
+        return physicalPartRepository.getObject(Integer.valueOf(physicalPartID));
+    }
+
+    @Override
+    public PhysicalPart createNewPhysicalPart(PhysicalPart physicalPart) {
+        return physicalPartRepository.postObject(physicalPart);
+    }
+
+    @Override
+    public PhysicalPart updatePhysicalPart(String physicalPartID, PhysicalPart physicalPart) {
+        checkIDs(physicalPartID, physicalPart);
+        return physicalPartRepository.updateObject(physicalPart);
+    }
+
+    @Override
+    public List<PhysicalPrint> getAllPhysicalPrints() {
+        List<PhysicalPrint> resultList = new LinkedList<PhysicalPrint>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)